### DISCUSSION TITLE
Moving Plots - Farming on 2 machines?

### DISCUSSION THREAD

**Discussion by scotchegg** [Chia-Network/chia-blockchain#1627] [General]
Hi 

I am moving plots from a faster machine to a slower machine while i fill the HD.
However while I usually delete them, today I noticed both machines appear to be farming the same plots before I do that? Both are under the same PKEY, and both report differing estimates to win with the correct plot count per machine. 

Is this a bug, would this work? Is this an exploit?

**Comment by ardabro** [comment]
I've just wanted to ask a similar question. 
If indeed many logical nodes may share the same plots on the same physical media it is the funniest bug I've ever seen :)

**Reply by dorkmo** [reply]
i'd think the plot would have the same chance of winning no matter how many farmers have access to it, probably whatever computer proves it first will be the one that registers it to the network

**Comment by scotchegg** [comment]
If I have 2 tb of plots on a network drive with seperate 10 machines pointing to the same plots, is this then a 2tb chance of winning ( what each machine reports, say 18 days) or a 20tb chance of winning and a 1.8 days chance?

@hoffmang9

**Comment by ardabro** [comment]
If the second option is true, this project is worthless

pt., 2 kwi 2021 o 08:56 scotchegg ***@***.***> napisał(a):

**Reply by hoffmang9** [reply]
The estimated time to win only uses total space and not total unique space. Therefore the estimate is wrong if you have duplicate plots.

**Reply by shock2d** [reply]
Hi @hoffmang9 
Even if these nodes are on separate machines and networks?

**Reply by SackofFluor** [reply]
As far as i experienced it just depends on the keys used to generate the plots. If you produce them with the same key you can just copy them to a harddrive, carry it home and plug copy them over.
What I read so far is though, that you should rename the plots for copying  by adding ".tmp" so only once the whole file is copied you rename it back and the farm should recognizes within a couple of minutes.

**Reply by room101-dev** [reply]
> As far as i experienced it just depends on the keys used to generate the plots. If you produce them with the same key you can just copy them to a harddrive, carry it home and plug copy them over.
> What I read so far is though, that you should rename the plots for copying by adding ".tmp" so only once the whole file is copied you rename it back and the farm should recognizes within a couple of minutes.

By default when you open up new machines on your lan and run the harvester  it creates it own ~/.chian/main-net folder, you have to make sure that your just running the daemon, and you need on a lan to point back in the config.yaml file to the system containing the full & real blockchan ( /db folder )
Plotting has NOTHING to do with the harvester, the plotter just makes .plot files and dumps them into the -d target.
The path to the .plot files a list of all folder is kept in the config.yaml file

Normally the simplest thing to do is dump your plots on machines on your LAN, and then run samba to share folders, then get the cmd-line path of that samba external, and paste that into the GUI, or the config.yaml of the your main-net invocation

The plotter will not write to samba, but the harvester can read .plots on your lan.

For me the harvester is SHIT, unless you have 10k plots your not going to win solo-mining, so I run hpool on all the machines that have plot-files and give each machine a unique name, hpool has its own .yaml file just like chia

The simplest thing to do is to  put drives containing plots on all your machines on your network, setup samba file sharing system, and then the GUI can autoatically find these shared network files and automatically setup the paths to the plots files all over your lan for you. But like I said, I prefer not to use the GUI. But after you have setup the samba and the chia gui has found, then you can look at the real samba names as used by chia, and remember that format, which then you can duplicate on all your external plots files on your lan.

Many ways to do this, harvesting plots with CHIA sw is a waste of time, you'll never earn XCH, you must run hpool to earn xch.

**Comment by rlallen-nps** [comment]
I was wondering the same question and found your post. I need to look at code, but am assuming plot generation is tied to your particular key. Have you discovered anything since posting?

**Comment by room101-dev** [comment]
> Hi
> 
> I am moving plots from a faster machine to a slower machine while i fill the HD.
> However while I usually delete them, today I noticed both machines appear to be farming the same plots before I do that? Both are under the same PKEY, and both report differing estimates to win with the correct plot count per machine.
> 
> Is this a bug, would this work? Is this an exploit?

You need to edit the file config.yaml, in ~/.chia/main-net/config, there you will see a list of directorys that are to be processed by the harvester, you can also do cmd-line "Chia plot show" to dump the dirs, but the file is the real deal

The problem most likely here is you moved the files to the new folder and/or machine on your land, but you didn't re-cache the sytsem, normally if you change the 'config.yaml', you need to run "chia start -r all" to get everything to use the new config.

The  harvester is going to look in all those dirs for plots and add them to a list, don't worry about doubles as its not dumb, it can see a duplicate.

The reality the final word is the contents of config.yaml, 99% of all problems here is people use the GUI, which is broken, and doesn't not provide a real view of the system.
